kernel/univ.ml

```diff
@@ -976,7 +976,9 @@ struct
 
   let instantiate inst (u, cst) =
     assert (Array.length u = Array.length inst);
-    subst_instance_constraints inst cst
```

Comment:
This seems to be the wrong place to enforce this to me. It's effectively hardwiring unrelated design choices inside a generic purpose function. Is it too hard to add those constraints at the time of the creation of the abstract context?